### DISCUSSION TITLE
How to move a single app to a separate new repository?

### DISCUSSION THREAD

**Discussion by MaurizioCasciano** [nrwl/nx#7736] [Feature Requests]
Let's consider an NX workspace with different _apps_ 
**A, B, C**
and _libs_
**LA, LB, LC, Shared**,
belonging to the private **App-Mono** git repository.

I am looking for the best way to opensource a subset of the monorepo, **App-C**, to the community.

1. How can we move the app **C**, together with **LC** and **Shared** libs (and the other dependencies, e.g. package.json)
to a separate/open-source git repository? Maybe keeping the NX Workspace configuration, but just for that particular app and its dependencies.

1. How to have this new repository linked with the original monorepo repository in order to easily merge back and forth the new commits from/to the community?

A possible approach I did find is to use the [Git Filter-Repo](https://github.com/newren/git-filter-repo) command
in order to create a new repository with only the needed files, but maybe there is another way to do it.

**Comment by AndriiTsok** [comment]
@MaurizioCasciano Google has a very handy tool for that https://github.com/google/copybara 

Would be great to hear if NX has any capabilities for a proper moving of an app out of a monorepo.

**Reply by MaurizioCasciano** [reply]
Thank you @AndriiTsok, do you use it with NX or do you have a tutorial?
I will surely investigate more on it.
I hope that the NX team could automate this feature.

**Reply by AndriiTsok** [reply]
@MaurizioCasciano nope, we do not have this set up yet. But we are also planning to open source some apps from our big monorepo.

**Reply by MaurizioCasciano** [reply]
Maybe creating a new branch and running a set of `@nrwl/workspace:remove` commands could be enough to have a new workspace with a single app and some libs.
https://nx.dev/l/a/workspace/remove

If all the commands are in a bash script then it would be just a matter of running a single command `npm run extract:app-C` delegating to the bash script.

What do you think?

@FrozenPandaz @vsavkin 
Would it be possible to list all the dependencies of e.g the **App C** and remove only the other projects not used by the App C in a single command, e.g `@nrwl/workspace:extract`?

**Reply by JulieMarie** [reply]
@MaurizioCasciano Were you able to accomplish this?  Either by using  `copybara` or by another way?

**Reply by MaurizioCasciano** [reply]
@JulieMarie as a quick solution I am creating a my-project-oss branch and running a set of 
`nx g @nrwl/workspace:remove <project-id>` on it.
The problem is that you need to write a script manually with the apps and libs to be removed.
Maybe there is some command to list all the apps and the dependencies of a project so that it could be automated, but I didn't find it. 
Do you know if/how it is possible to list all the apps of a nx-workspace, and the dependencies projects of an app?

**Comment by IlanSapiens** [comment]
Hi [MaurizioCasciano](https://github.com/MaurizioCasciano) did you find solution or got progress with that?
